modules/pubCommonIdSystem.js

```diff
@@ -8,20 +8,135 @@
 import * as utils from '../src/utils.js';
 import {submodule} from '../src/hook.js';
 import {getStorageManager} from '../src/storageManager.js';
+import {ajax} from '../src/ajax.js';
 
 const PUB_COMMON_ID = 'PublisherCommonId';
-
 const MODULE_NAME = 'pubCommonId';
 
+const COOKIE = 'cookie';
+const LOCAL_STORAGE = 'html5';
+const SHAREDID_OPT_OUT_VALUE = '00000000000000000000000000';
+const SHAREDID_URL = 'https://id.sharedid.org/id';
+const SHAREDID_SUFFIX = '_sharedid';
+const EXPIRED_COOKIE_DATE = 'Thu, 01 Jan 1970 00:00:01 GMT';
+
 const storage = getStorageManager(null, 'pubCommonId');
 
+/**
+ * Store sharedid in either cookie or local storage
+ * @param {Object} config Need config.storage object to derive key, expiry time, and storage type.
+ * @param {string} value Shareid value to store
+ */
+
+function storeData(config, value) {
+  if (value) {
+    const key = config.storage.name + SHAREDID_SUFFIX;
+    if (config.storage.type === COOKIE) {
+      if (storage.cookiesAreEnabled()) {
+        const expiresStr = (new Date(Date.now() + (storage.expires * (60 * 60 * 24 * 1000)))).toUTCString();
+        storage.setCookie(key, value, expiresStr, 'LAX', COOKIE_DOMAIN);
+      }
+    } else if (config.storage.type === LOCAL_STORAGE) {
+      if (storage.hasLocalStorage()) {
+        storage.setDataInLocalStorage(key, value);
+      }
+    }
+  }
+}
+
+/**
+ * Read sharedid from cookie or local storage
+ * @param config Need config.storage to derive key and storage type
+ * @return {string}
+ */
+function readData(config) {
+  const key = config.storage.name + SHAREDID_SUFFIX;
+  if (config.storage.type === COOKIE) {
+    if (storage.cookiesAreEnabled()) { return storage.getCookie(key); }
+  } else if (config.storage.type === LOCAL_STORAGE) {
+    if (storage.hasLocalStorage()) {
+      return storage.getDataFromLocalStorage(key);
+    }
+  }
+}
+
+/**
+ * Delete sharedid from cookie or local storage
+ * @param config Need config.storage to derive key and storage type
+ */
+function delData(config) {
+  const key = config.storage.name + SHAREDID_SUFFIX;
+  if (config.storage.type === COOKIE) {
+    if (storage.cookiesAreEnabled()) {
+      storage.setCookie(key, '', EXPIRED_COOKIE_DATE);
+    }
+  } else if (config.storage.type === LOCAL_STORAGE) {
+    storage.removeDataFromLocalStorage(key);
+  }
+}
+
+/**
+ * setup success and error handler for sharedid callback thru ajax
+ * @param {string} pubcid Current pubcommon id
+ * @param {function} callback userId module callback.
+ * @param {Object} config Need config.storage to derive sharedid storage params
+ * @return {{success: success, error: error}}
+ */
+
+function handleResponse(pubcid, callback, config) {
+  return {
+    success: function (responseBody) {
+      if (responseBody) {
+        try {
+          let responseObj = JSON.parse(responseBody);
+          utils.logInfo('SharedId: Generated SharedId: ' + responseObj.sharedId);
```

Comment:
Should this be more like this, so that it's clear this message is coming from the pubcommon module and not shared id?

```suggestion
          utils.logInfo('PubCommonId: Generated SharedId: ' + responseObj.sharedId);
```